package/MDAnalysis/coordinates/MMTF.py

```diff
@@ -77,7 +79,9 @@ def _read_first_frame(self):
         ts._pos[:, 0] = top.x_coord_list
         ts._pos[:, 1] = top.y_coord_list
         ts._pos[:, 2] = top.z_coord_list
-        ts.dimensions = top.unit_cell
+        if not top.unit_cell is None:
```

Comment:
Is  `ts.dimensions` that is not defined my MMTF set to something sensible? Do we not require Timestep to have unitcell defined? Will bad things happen if a ts does not have it?

Reply:
So it gets set to `None` now.  So calling things like `distance_array(a, b, box=u.dimensions)` will proceed silently without PBC, which is maybe wrong as the user intended there to be a box?  Could instead set to [0, 0, 0] which would make distance calculations scream (hopefully)

Reply:
@richardjgowers You're right, `None` will disable PBC. In `lib.distances`, all-zeros dimensions will run into `lib.util.check_box()` and will currently be interpreted as a triclinic cell since not all angles are 90 degrees. That will trigger calls to the respective triclinic C implementation, which will commonly result in a no-op for most of the routines. So, either way, things *will* go wrong.

A quick-and-dirty solution to the problem might be making `lib.util.check_box()` throw an error for all-zero boxes.

What we *really* need is a standardized and consistently enforced way to discriminate between no PBC and missing information!

Reply:
BTW that also affects PR #2201

Reply:
Tbh the really nice solution is to have 0 dimensions to be non periodic, so we could have semi periodic PBC...

Maybe `u.dimensions` should raise `AttributeError` if there's no box, it's essentially a missing attribute

Reply:
>Maybe `u.dimensions` should raise `AttributeError` if there's no box, it's essentially a missing attribute

I like this idea. That's completely in line with #2199 btw.
>Tbh the really nice solution is to have 0 dimensions to be non periodic, so we could have semi periodic PBC...

Semi-periodic PBC sounds awesome, but has to be taken with a grain of salt. We cannot do this efficiently with the current state of the PBC-related C-level implementations since that would require introducing branching in hot loops. I'm afraid that hoping for branch prediction to keep us up to speed won't work out here because branching will most likely prevent compiler optimizations such as loop auto-vectorization an the like.

Reply:
Forget the last part of what I just wrote. Semi-periodic PBC should already work.

Reply:
Semi-periodic boxes cannot be indicated by putting a zero value in the non-periodic direction. In general, that only works for true 2d systems. For PBC wrapping this might also work in 3d (not sure about triclinic boxes, though), but it might break pretty much anything else in semi-periodic 3d syatems, such as box volume calculation and so on.

Reply:
So what's the conclusion for here? Wait for `AttributeError` to show up in strange places or use `None`?

We could guess a bounding box if we really wanted to but this seems too much magic under the hood (so I am _not_ in favor of guessing).

Reply:
Sorry, shoul've looked at new diff: `None`.